neo/Consensus/ConsensusContext.cs

```diff
@@ -174,6 +194,31 @@ private ConsensusPayload MakeSignedPayload(ConsensusMessage message)
             return payload;
         }
 
+        public ConsensusPayload RegenerateSignedPayload(ConsensusMessage message, ushort validatorIndex,
+            byte[] witnessInvocationScript, uint timestamp)
+        {
+            message.ViewNumber = ViewNumber;
```

Comment:
For a `ChangeView` message, the `viewnumber` is not necessarily the current `viewnumber`.

I suggest changing this method to:

```c#
public ConsensusPayload RegenerateSignedPayload(ConsensusMessage message, ConsensusPayload payload, ushort validatorIndex)
```

Reply:
I changed this to not set the `ViewNumber`

Reply:
I had left out the file that removed setting the view number when I made this initial comment 22 minutes ago. I just pushed the change now.

Reply:
The view number of ChangeView message can not be ignored. Because it is a part of the data to be signed.

Reply:
@erikzhang the latest changes on this PR already handle it. The `ViewNumber` for all change view messages is set to 0, because no code uses the view number it only uses the NewViewNumber. If we really want it to stay as the sender's view number we can save those to in the recovery message, but it is unnecessary IMO.

Reply:
If you don't set it to it's original value, how to verify the signature?

Reply:
@erikzhang 
The original value is always 0. Just use 0 and it verifies perfectly.

Reply:
Specifically the code change that switched the original value for ChangeView to always be 0 is here (from about 3 hours ago): c43beff

Reply:
For the messages that are not ChangeView, it sets them to their original value which will always be the same as the Recovery message value. For the ones that are ChaneView, it sets them to 0 which is what their original value will always be now.